lib/secrex/aes.ex

```diff
@@ -30,13 +30,7 @@ defmodule Secrex.AES do
     init_vector = initialize_vector(@iv_length)
     key_digest = hash(key)
 
-    {encrypted, tag} =
-      :crypto.block_encrypt(
-        :aes_gcm,
-        key_digest,
-        init_vector,
-        {@aad, plaintext, @tag_length}
-      )
+    {encrypted, tag} = :crypto.crypto_one_time_aead(:aes_gcm, key_digest, init_vector, plaintext, @aad, @tag_length, true)
```

Comment:
We would still need to support OTP versions before 22 (when this function was introduced).
This means a conditional compilation, for example: https://github.com/lexmag/oauther/commit/b5ad3bdeca1916a41eec51335947ed1599e20c76.

Reply:
Done!